File: src/sql/postgres/Status.zig
pub const Status = enum {
    disconnected,
    connecting,
    // Prevent sending the startup message multiple times.
    // Particularly relevant for TLS connections.
    sent_startup_message,
    connected,
    failed,

    pub fn hasPendingActivity(this: Status) bool {
        return switch (this) {
            .connected => false,
            .connecting => true,
            .sent_startup_message => true,
            .failed => false,
            .disconnected => false,
        };
    }
};

File: src/sql/postgres/Status.rs
#[derive(Copy, Clone, Eq, PartialEq, Debug)]
pub enum Status {
    Disconnected,
    Connecting,
    // Prevent sending the startup message multiple times.
    // Particularly relevant for TLS connections.
    SentStartupMessage,
    Connected,
    Failed,
}

// ported from: src/sql/postgres/Status.zig
